build with USE_GPL_LIBS=0 fails with UndefVarError(:CHOLMOD)

My last successful build was at f965fd4.
```
/Users/sachs/src/julia-master/base/precompile.jl
LoadError("sysimg.jl", 858, LoadError("/Users/sachs/src/julia-master/base/precompile.jl", 19, UndefVarError(:CHOLMOD)))
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
make[1]: *** [/Users/sachs/src/julia-master/usr/lib/julia/sys.o] Error 1
make: *** [julia-sysimg-release] Error 2
```

--- a/base/precompile.jl
+++ b/base/precompile.jl
@@ -874,7 +874,9 @@ precompile(Tuple{typeof(REPL.send_to_backend), String, Base.Channel{Any}, Base.C
 precompile(Tuple{typeof(REPL.send_to_backend), String, REPL.REPLBackendRef})
 precompile(Tuple{typeof(Random.DSFMT.__init__)})
 precompile(Tuple{typeof(Random.__init__)})
-precompile(Tuple{typeof(SuiteSparse.CHOLMOD.__init__)})
+if Base.USE_GPL_LIBS
+    precompile(Tuple{typeof(SuiteSparse.CHOLMOD.__init__)})
+end
 precompile(Tuple{typeof(Test.get_test_result), Expr, LineNumberNode})
 precompile(Tuple{typeof(Test.get_test_result), Symbol, LineNumberNode})
 precompile(Tuple{typeof(Test.test_expr!), String, Expr})

--- a/contrib/fixup_precompile.jl
+++ b/contrib/fixup_precompile.jl
@@ -1,3 +1,8 @@
+function needs_USE_GPL_LIBS(s::String)
+    contains(s, "CHOLMOD") && return true
+    return false
+end
+
 const HEADER = """
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
@@ -17,7 +22,8 @@ function fixup_precompile(new_precompile_file; merge=false)
 
     for file in [new_precompile_file; merge ? old_precompile_file : []]
         for line in eachline(file)
-            # filter out closures, which might have different generated names in different environments)
+            line = strip(line)
+            # filter out closures, which might have different generated names in different environments
             contains(line, r"#[0-9]") && continue
             # Other stuff than precompile statements might have been written to STDERR
             startswith(line, "precompile(Tuple{") || continue
@@ -38,7 +44,10 @@ function fixup_precompile(new_precompile_file; merge=false)
         end
         @eval PrecompileStagingArea begin""")
         for statement in sort(collect(precompile_statements))
+            isgpl = needs_USE_GPL_LIBS(statement)
+            isgpl && print(f, "if Base.USE_GPL_LIBS\n    ")
             println(f, statement)
+            isgpl && println(f, "end")
         end
         println(f, "end\nend")
     end

